Automattic/wp-calypso
# Emails: Fix undefined email placeholder label on add mailbox forms

#### Changes proposed in this Pull Request

The fix for the issue 1200245279504217-as-1201117888572949 added a regression on adding mailboxes, where the label to show the domain is being undefined. This code fix it.

#### Testing instructions

**Test 1**

1. Go to Upgrade -> Emails
2. Add a mailbox for Professional Email
3. Check that there is no undefined label in the email input
![image](https://user-images.githubusercontent.com/5689927/138877625-62d47a2a-d0c7-473b-bf96-c79e1ac61258.png)

**Test 2**

1. Go to Upgrade -> Emails
2. Add a mailbox for Google Workspace
3. Check that there is no undefined label in the email input

**Test 3**

Remove mailboxes from previous tests, for both subscriptions and click on the "Setup mailbox" button
![image](https://user-images.githubusercontent.com/5689927/138877776-cc465db0-1b7d-4f1f-ac87-a3d0185bf953.png)

**Test 4**
[Upgrades]->[Domains]->[Add a domain to this site]->[Search for a domain]

**Test 5**
Having 2 domains without emails, being in the comparison page switch between those two sites and check that it updates the URL, as per the test in this [pull request](https://github.com/Automattic/wp-calypso/pull/57029)

Related to:

Regression: 1200245279504217-as-1201117888572949
Task: 1200182182542585-as-1201270706976018
